src/test/java/uk/gov/pay/directdebit/mandate/services/MandateStateUpdateServiceTest.java

```diff
@@ -148,26 +172,13 @@ public void mandatePendingFor_shouldRegisterAMandatePendingEvent() {
                 .aMandateFixture()
                 .withState(SUBMITTED)
                 .toEntity();
+        Mandate expectedMandateWithPendingState = fromMandate(mandate)
+                .withState(PENDING)
+                .build();
 
         service.mandatePendingFor(mandate);
 
-        verify(mockedDirectDebitEventService).registerMandatePendingEventFor(mandate);
-        assertThat(mandate.getState(), is(PENDING));
-    }
-
-    @Test
-    public void shouldThrowWhenSubmittingTheSameDetailsSeveralTimes_andCreateOnlyOneEvent_whenConfirmingDirectDebitDetails() {
```

Comment:
Why was this test removed?

Reply:
The test never worked as expected since lines 169 and 170 we're never being invoked; execution stopped after checking the thrown exception as a result of line 167.

Now that we don't mutate mandates, it doesn't make sense to invoke `service.confirmedOnDemandDirectDebitDetailsFor(onDemandMandate);` with the same mandate object twice, to confirm it will throw an exception, since the immutable mandate will keep on passing. There is a test that confirms we cannot confirm a mandate that is in the state of `DIRECT_DEBIT_DETAILS_CONFIRMED`.

Reply:
I have added an assertion to ensure `registerDirectDebitConfirmedEventFor` is being called when confirming a mandate within test `shouldUpdateMandateStateAndRegisterEventWhenConfirmingDirectDebitDetails_andSendEmail`